internal/bminventory/inventory.go

```diff
@@ -291,15 +292,17 @@ func (b *bareMetalInventory) UpdateDiscoveryIgnitionInternal(ctx context.Context
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
 
-	b.eventsHandler.AddEvent(ctx, params.ClusterID, nil, models.EventSeverityInfo, "Custom discovery ignition config was applied to the cluster", time.Now())
+	ev := eventgen.NewClusterDiscoveryIgnitionConfigAppliedEvent(params.ClusterID)
+	b.eventsHandler.SendClusterEvent(ctx, ev)
```

Comment:
This pattern is repeated everywhere. Can we do something to make it a one line operation? Maybe add an events handler as a parameter to the generated functions and send the event from there?

Is there ever a use case to create an event and not send it?

Reply:
+1

Reply:
Replaced this pattern.
Can you please re-review it? There are few examples in the updated PR.